### PR TITLE
release: add PodNotFound condition

### PR DESCRIPTION
**Pull request by kopiczko**
Towards https://github.com/giantswarm/giantswarm/issues/5278.

### CODE REVIEW

File: pkg/release/condition_set.go [hunk 1]
```diff
@@ -78,7 +78,7 @@ func (c *conditionSet) CRDNotFound(ctx context.Context, crd *apiextensionsv1beta
 				return backoff.Permanent(microerror.Mask(err))
 			}
 
-			return microerror.Maskf(waitError, "CRD %#q in namespace %#q still exists", crd.Name, crd.Namespace)
+			return microerror.Maskf(waitError, "CRD %#q still exists", crd.Name, crd.Namespace)
```
Comment (MarcelMue) [inline pkg/release/condition_set.go]: How does this look like now?

File: pkg/release/condition_set.go [hunk 2]
```diff
@@ -120,6 +120,32 @@ func (c *conditionSet) PodExists(ctx context.Context, namespace, labelSelector s
...
+			if apierrors.IsNotFound(err) {
+				return nil
+			} else if err != nil {
+				return microerror.Mask(err)
+			}
+			if len(pods.Items) != 1 {
+				return microerror.Maskf(waitError, "expected 1 Pod but got %d", len(pods.Items))
+			}
+
+			return microerror.Maskf(waitError, "Pod for label selector %#q in namespace %#q still exists", labelSelector, namespace)
```
Comment (MarcelMue) [inline pkg/release/condition_set.go]: Here you still use the namespace in the Maskf

File: pkg/release/condition_set.go (hunk 1 above)
Comment (kopiczko) [reply pkg/release/condition_set.go]: CRD is cluster scope. I saw it somewhere but I don't remember.

File: pkg/release/condition_set.go (hunk 2 above)
Comment (kopiczko) [reply pkg/release/condition_set.go]: Pod is namespaced :)

File: pkg/release/condition_set.go (hunk 1 above)
Comment (MarcelMue) [reply pkg/release/condition_set.go]: Then remove the `crd.Namespace` from here :D

Comment (kopiczko) [reply pkg/release/condition_set.go]: Ah ok :)
```suggestion
			return microerror.Maskf(waitError, "CRD %#q still exists", crd.Name)
```

**Review by MarcelMue — APPROVED** [review]

**Pull request merged**